src/Item.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 - present, Facebook
+
+#include "iterlib/Item.h"
+
+namespace iterlib {
+
+const std::string kIdKey(QueryConstants::kIdKey);
+const std::string kTypeKey(QueryConstants::kTypeKey);
+const std::string kTimeKey(QueryConstants::kTimeKey);
+
+namespace {
+const dynamic kEmptyItemInternal{variant::unordered_map_t{
+    {{":id", int64_t(std::numeric_limits<id_t>::max())}, {":time", 0}}}};
+}
+
+const Item Item::kEmptyItem{kEmptyItemInternal};
+
+PartialOrder partialCompare(const Item& v1, const Item& v2,
```

Comment:
Might be good to add some documentation explaining a parallel to SQL
`ORDER BY column_name ASC|DESC, column_name ASC|DESC; `